src/rest_api/api/search/substructure_search.rs

```diff
@@ -49,9 +51,38 @@ pub fn v1_index_search_substructure(
 
     let (query_canon_taut, fingerprint, descriptors) = query_attributes;
 
+    let scaffolds = if use_scaffolds {
+        let all_scaffolds = get_scaffolds();
+        match all_scaffolds {
+            Ok(all_scaffolds) => all_scaffolds,
+            Err(e) => {
+                return GetStructureSearchResponse::Err(Json(StructureResponseError {
+                    error: e.to_string(),
+                }))
+            }
+        }
+    } else {
+        Vec::new()
+    };
```

Comment:
Could this logic be redone with a `Option<Vec<_>>` instead of passing around an empty vec?